community/kernel/src/main/java/org/neo4j/kernel/impl/storemigration/UpgradeNotAllowedByConfigurationException.java

```diff
@@ -25,14 +25,19 @@
 {
     public UpgradeNotAllowedByConfigurationException( String msg )
     {
-        super( msg );
+        super( String.format( "%s Detailed description: %s", baseMessage() , msg)  );
     }
 
     public UpgradeNotAllowedByConfigurationException()
     {
-        super( String.format(
-                "Failed to start Neo4j with an older data store version. "
-                        + "To enable automatic upgrade, please set configuration parameter \"%s=true\"",
-                GraphDatabaseSettings.allow_upgrade.name() ) );
+        super( baseMessage() );
+    }
+
+    private static String baseMessage()
+    {
+        return String.format(
+                "Neo4j cannot be started, because the database files require upgrading and upgrades are " +
```

Comment:
`upgrades are`, dunno about the plural there. Perhaps `upgrading is`... just taking extra care since it's public surface. WDYT?